contracts/LockScheme.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./interfaces/IERC20Detailed.sol";
+import "./SafeERC20Detailed.sol";
+import "./PercentageCalculator.sol";
+
+contract LockScheme {
+
+    using SafeMath for uint256;
+    using SafeERC20Detailed for IERC20Detailed;
+
+    uint256 public lockBlock; // The period of lock for this contract
+    uint256 public rampUpBlock; // The period since the beginning of the lock that additions can be considered the same position.Might be 0 for the 0% lock periods
+    address public stakingToken;
+    uint256 public bonusPercent; // saved in thousands = ex 3% = 3000
+    address public lmcContract; // The address of the lmc contract
+    uint256 public forfeitedBonuses;
+
+    struct UserInfo {
+        uint256 balance; // IOU Balance for this lock contract
+        uint256[] accruedReward; // Reward accrued by an address from previous additions
```

Comment:
The lock bonus is only on the ALBT. Therefore you do not need an array of rewards